pkg/command/map_string.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of Cilium
+
+package command
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cast"
+	"github.com/spf13/viper"
+)
+
+var keyValueRegex = regexp.MustCompile(`((\w+)=(\w+,))+((\w+)=(\w+))$`)
+
+// GetStringMapString contains one enhancement to support k1=v2,k2=v2 compared to original
+// implementation of GetStringMapString function
+func GetStringMapString(vp *viper.Viper, key string) map[string]string {
+	v, _ := GetStringMapStringE(vp, key)
+	return v
+}
+
+// GetStringMapStringE is same as GetStringMapString, but with error
+func GetStringMapStringE(vp *viper.Viper, key string) (map[string]string, error) {
+	data := vp.Get(key)
+	v, err := cast.ToStringMapStringE(data)
+	if err != nil {
+		var syntaxErr *json.SyntaxError
+		if !errors.As(err, &syntaxErr) {
+			return v, err
+		}
+
+		switch s := data.(type) {
+		case string:
+			// if the input is starting with either '{' or '[', just preserve original json parsing error.
+			firstIndex := firstNonSpaceCharIndex(s)
+			if firstIndex != -1 && (s[firstIndex] == '{' || s[firstIndex] == '[') {
+				return v, err
+			}
+
+			if !isValidKeyValuePair(s) {
+				return map[string]string{}, fmt.Errorf("'%s' is not formatted as key=value,key1=value1", s)
+			}
+
+			var v = map[string]string{}
+			kvs := strings.Split(s, ",")
+			for _, kv := range kvs {
+				temp := strings.Split(kv, "=")
+				if len(temp) != 2 {
+					return map[string]string{}, fmt.Errorf("'%s' is not formatted as key=value,key1=value1", s)
+				}
```

Comment:
ideally, this code block is not required as above isValidKeyValuePair fucntion should cover for invalid format, it's just to be safe to have this check.